### PR TITLE
Engine FIM decoder

### PR DESCRIPTION
**Pull request by nmkoremblum**
|Related issue|
|---|
|#13524|

## Description

As requested on Issue #13524, a way of decoding FIM events must be added to the engine. This would require creating its YAML-formatted decoder and complementary helper functions if needed.

Additionally, some helper functions to check a JSON field's type were added, as the `is_number` one was required to develop the FIM decoder.

### CODE REVIEW

**Review by LucioDonda — APPROVED** [review]
LGTM

**Review by juliancnn — APPROVED** [review]
LGTM!

File: src/engine/source/builder/builders/opBuilderHelperFilter.cpp [hunk 1]
```diff
@@ -791,4 +792,430 @@ base::Expression opBuilderHelperContainsString(const std::any& definition)
...
+            if (event->exists(targetField) && event->isNumber(targetField))
+            {
+                return base::result::makeSuccess(event, successTrace);
+            }
+            else
+            {
+                return base::result::makeFailure(event, failureTrace);
+            }
+        });
+}
```
Comment (JavierBejMen) [inline src/engine/source/builder/builders/opBuilderHelperFilter.cpp:827]: We should give a concise error trace of why it failed:

```c++
if (!event->exists(targetField)
{ // Error, target field not found
...
}
```
This applies to all added helpers

**Review by JavierBejMen — CHANGES_REQUESTED** [review]

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**